theano/tensor/tests/test_opt.py

```diff
@@ -1997,10 +1997,10 @@ def test_subtensor_inc_subtensor():
     mode = theano.compile.mode.get_default_mode().including('local_subtensor_inc_subtensor')
     f = theano.function([x, i1, i2, v], z, mode=mode)
     prog = f.maker.fgraph.toposort()
-    assert any(isinstance(x.op, tensor.Alloc) for x in prog)
+    assert any(isinstance(x.op, (tensor.Alloc, theano.sandbox.cuda.basic_ops.GpuAlloc)) for x in prog)
```

Comment:
remove GpuAlloc. The CPU tests should not request the gpu back-end. This would also not work with the new back-end.

I think the problem is that  you had device⁼gpu* Tests suppose device=cpu. So don't try to support device!=cpu, too many tests already fail in that case. Also, currently I think this would fail when device=cpu, as theano.sandbox.cuda isn't imported.

If for some reason, you still need to keep GpuAlloc, add import theano.sandbox.cuda before using this, not at the top of the file.